pkg/util/wait_group_wrapper.go

```diff
@@ -29,17 +29,20 @@ import (
 // if the `exited` signal is true by print them on log.
 type WaitGroupEnhancedWrapper struct {
 	sync.WaitGroup
-	source          string
-	registerProcess sync.Map
+	source string
+	mu     struct {
```

Comment:
If we configure this in the tidb toml configuration file

```
tidb-enable-exit-check = true
```

We can get the unexited services ... but it's not seems to be enabled by default.

So anyway, we have `dlv`, we can attach the tidb process to see where it blocks.
We can checkout this goroutine stack, and print the value of `registerProcess` to see which services are unexited.

The old code use `sync.Map`, which is opaque, not debuggable.
Change it to map is much easier for debugging.

I don't think the performance really matters here, so mutex is prefered than `sync.Map`.